okhttp-clients/src/main/java/com/palantir/conjure/java/QosExceptionResponseMapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java;
+
+import com.google.common.net.HttpHeaders;
+import com.palantir.conjure.java.api.errors.QosException;
+import com.palantir.logsafe.UnsafeArg;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class QosExceptionResponseMapper {
+
+    private static final Logger log = LoggerFactory.getLogger(QosExceptionResponseMapper.class);
+
+    private QosExceptionResponseMapper() {
+    }
+
+    public static Optional<QosException> mapResponseCodeHeaderStream(
```

Comment:
Bit weird that this is technically becoming public API...

Reply:
I agree. We could get around this by moving all of its consumers into one package and making it package private.